packages/java-parser/src/tokens.js

```diff
@@ -203,8 +203,7 @@ createToken({
 });
 createToken({
   name: "StringLiteral",
-  // TODO: align with the spec, the pattern below is incorrect
-  pattern: /"[^"\\]*(\\.[^"\\]*)*"/
+  pattern: /"(?:[^\\"]|\\(?:(?:[btnfr"'\\/]|[0-7]|[0-7]{2}|[0-3][0-7]{2})|u[0-9a-fA-F]{4}))*"/
```

Comment:
can we re-use the regexp from charLiteral instead of copy/paste?
Have a look at some MAKE_PATTERN examples in this file

Reply:
I was actually wondering if it was worth it to rename most of the regexp using the java specs and fragments/make_patterns to make it easier to read and translate directly from the java specs. But I am unsure if it has huge performance impacts.

We could make small PR's bit by bit to slowly use fragments

Reply:
It is generally a good thing.
Note I implemented a better fragments composition system here:
- https://github.com/bd82/toml-tools/blob/master/packages/lexer/lib/tokens.js
- I think I was also trying to align with the TOML spec names for the lexer fragments there.

It may not be worth the effort to do so now, depends on how much time resources we have...
But we should at a minimum avoid new duplications...

Reply:
Yes I will try to do it then. Should we merge this one then do a later PR for fragments or should I just commit there?

Reply:
I think that a big refactor for using more fragments should be done **separately**.

But that this PR for the stringLiteral + char patterns should avoid the **new** duplication.
So you should define a single fragment for re-use in char/stringliteral as part of this PR.

Basically do not introduce **more** technical debt in this PR
And consider resolving the existing technical debt at a later time depending on available time resources.

Reply:
I've managed to re-do the Character and String literals tokens. It was a bit tricky as MAKE_PATTERN transform \\ into \ so I had to go for quad-\'s ...

Reply:
Yeah, the horrors of multiple levels of escaping 😠